web/src/compose_banner.ts

```diff
@@ -61,11 +61,34 @@ export function get_compose_banner_container($textarea: JQuery): JQuery {
 // This function provides a convenient way to add new elements
 // to a banner container. The function accepts a container element
 // as a parameter, to which a banner should be appended.
+// Returns a boolean value indicating whether the append had succeeded.
+// Cases where it would fail: when trying to append a warning banner
+// when a "no post permissions" error banner is already present.
 export function append_compose_banner_to_banner_list(
     banner: HTMLElement | JQuery.htmlString,
     $list_container: JQuery,
-): void {
+): boolean {
+    // Skip rendering warning banners if the user does not have post permissions.
+    if (container_has_post_permissions_error_banner($list_container)) {
+        if (typeof banner === "string") {
+            // Filter out warning banners from HTML string.
+            const parsed_nodes = $.parseHTML(banner.trim());
+            let new_banner = "";
+            for (const node of parsed_nodes) {
```

Comment:
In my opinion, it would be beneficial to rely on the idea that append_compose_banner_to_banner_list can add only one banner per call. Currently, we don't have any situations where we need to add multiple banners, and I don't anticipate that changing anytime soon. Thus, keeping the code simple seems like a good approach for now.

Additionally, if we ever do require multiple banners, we can easily implement a loop that calls append_compose_banner_to_banner_list as needed. I'm not overly concerned about performance since I don't expect there to be a large number of banners.

What do you think?

Reply:
I do agree! I was pondering whether to account for multiple banners or not. But it definitely makes sense to get rid of that assumption to make the code more readable.

I'm thinking of adding a check condition to make sure we actually do only have one top-level element in the HTML string input, just to make sure nothing unexpected comes up, since anything can be passed in. Do you think this is necessary?

Reply:
> I'm thinking of adding a check condition to make sure we actually do only have one top-level element in the HTML string input, just to make sure nothing unexpected comes up, since anything can be passed in. Do you think this is necessary?

While most of the instances where we call `append_compose_banner_to_banner_list` are clear and only involve adding a single element, adding a reliability check could be a good idea, IMO.

In case you believe this check might affect the readability of `append_compose_banner_to_banner_list`, it might be worth considering extracting it into a separate function.